src/JsonSchema/SchemaFactoryInterface.php

```diff
@@ -26,4 +26,6 @@ interface SchemaFactoryInterface
      * Builds the JSON Schema document corresponding to the given PHP class.
      */
     public function buildSchema(string $className, string $format = 'json', string $type = Schema::TYPE_OUTPUT, ?string $operationType = null, ?string $operationName = null, ?Schema $schema = null, ?array $serializerContext = null, bool $forceCollection = false): Schema;
+
+    public function addDistinctFormat(string $format): void;
```

Comment:
Adding a method to an existing interface is a BC break, you cannot do it.

Reply:
The problem here is that we decorating the hydra schema factory with the hal schemafactory. So the schemafactory property in the hal schema factory will never be a "BaseSchemaFactory".

We've got HalSchemaFacotry contains the HydraSchemaFactory contains the BaseSchemaFactory. Since you only could add distinct formats in the baseSchemaFactory, it is impossible for the HalSchemaFactory to add the 'hal' as a distinct format.

Do you've got any tips to add the hal distinct format without extending the schemafactory interface?

Reply:
You can keep the public method in the `HydraSchemaFactory` but without adding it to the interface too.

Reply:
Good point, I did work now with the method_exists function.